### PR TITLE
Use Infura IPFS for nft rewards images

### PR DESCRIPTION
**Pull request by tomquirk**
## What does this PR do and why?

As per title. This might break something to do with nft reward images. If it does, pls revert this PR.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by tomquirk** [comment]
cc @peripheralist

File: src/components/NftRewards/NftRewardsSection.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ function RewardTiersLoadingSkeleton() {
 
   return (
     <Row style={{ marginTop: '15px' }} gutter={isMobile ? 8 : 24}>
-      {[...Array(MAX_NFT_REWARD_TIERS)]?.map(index => (
+      {[...Array(MAX_NFT_REWARD_TIERS)]?.map((_, index) => (
```
Comment (tomquirk) [inline src/components/NftRewards/NftRewardsSection.tsx:34]: random fix

File: src/constants/ipfs.ts [hunk 2]
```diff
@@ -1,22 +1,7 @@
-import { NetworkName } from 'models/network-name'
-import { readNetwork } from './networks'
-
-const GOERLI_IPFS_OPEN_GATEWAY = 'jbx-goerli.mypinata.cloud' // restricted to goerli.juicebox.money.
```
Comment (tomquirk) [inline src/constants/ipfs.ts:4]: burn it

File: src/hooks/NftRewards.ts [hunk 3]
```diff
@@ -22,10 +20,12 @@ async function getRewardTierFromIPFS({
   tier: JB721TierParams
   index: number
 }): Promise<NftRewardTier> {
-  const url = restrictedIpfsUrl(decodeEncodedIPFSUri(tier.encodedIPFSUri))
+  const tierCid = decodeEncodedIPFSUri(tier.encodedIPFSUri)
+  const url = publicIpfsUrl(tierCid)
 
-  const response = await axios.get(url)
+  const response = await ipfsGet(url)
```
Comment (tomquirk) [inline src/hooks/NftRewards.ts:26]: get the NFT reward tier metadata using the infura endpoint (with the Authorization header)

File: .example.env [hunk 4]
```diff
@@ -20,6 +20,10 @@ NEXT_PUBLIC_BASE_URL=
 # sentry url. Not needed for development
 NEXT_PUBLIC_SENTRY_DSN=
 
+NEXT_PUBLIC_INFURA_IPFS_PROJECT_ID=
+NEXT_PUBLIC_INFURA_IPFS_API_SECRET=
+NEXT_PUBLIC_INFURA_IPFS_HOSTNAME=
```
Comment (peripheralist) [inline .example.env:25]: perfect

File: src/lib/infura/ipfs.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  headers: {
+    Authorization:
+      'Basic ' +
+      Buffer.from(
+        `${INFURA_IPFS_PROJECT_ID}:${INFURA_IPFS_API_SECRET}`,
+      ).toString('base64'),
+  },
+})
+
+export function ipfsGet(url: string) {
```
Comment (peripheralist) [inline src/lib/infura/ipfs.ts:18]: niiiice

File: src/utils/ipfs.ts [hunk 6]
```diff
@@ -22,7 +22,7 @@ export const metadataNameForHandle = (handle: string) =>
  * Return a URL to the public (open) IPFS gateway for the given cid.
  */
 export const publicIpfsUrl = (cid: string | undefined): string => {
-  return ipfsGatewayUrl(cid, OPEN_IPFS_GATEWAY_HOSTNAME)
+  return ipfsGatewayUrl(cid, OPEN_IPFS_GATEWAY_HOSTNAME as string)
```
Comment (peripheralist) [inline src/utils/ipfs.ts:25]: if we just need to type this as not-undefined, could do `OPEN_IPFS_GATEWAY_HOSTNAME!`

Comment (tomquirk) [reply src/utils/ipfs.ts:25]: totally, great point, even better :rock

**Pull request merged**